Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -2605,6 +2605,19 @@ - (void)clearUnreadMessagesIndicator
 {
     OWSAssertIsOnMainThread();
 
+    NSIndexPath *_Nullable indexPathOfUnreadIndicator = [self indexPathOfUnreadMessagesIndicator];
+    if (indexPathOfUnreadIndicator) {
+        ConversationViewItem *oldIndicatorItem = [self viewItemForIndex:indexPathOfUnreadIndicator.row];
+        OWSAssert(oldIndicatorItem);
+
+        // TODO ideally this would be happening within the *same* transaction that caused the unreadMessageIndicator
+        // to be cleared.
```

Comment:
Hmm, it seems like we could modify `[OWSMessageSender enqueueMessage]` to take an "enqueuedBlock" which would let us do some extra work in the same transaction.  NBD Just thinking out loud.